### PR TITLE
Leader Board : Shows Top Play Contributor of the month and Top Play Contributors of all time

### PR DESCRIPTION
**Pull request by NagarjunShroff**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Shows Top Play Contributor of the month and Top Play Contributors of all time

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshots or example output

image

### CODE REVIEW

File: src/common/playleaderboard/LeaderBoard.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import { useState, useEffect, useCallback } from 'react';
+import { submit } from '../services/request';
+import { FetchPlayCountByUser } from 'common/services/request/query/fetch-leaderboard-data';
+import TopPlayCreatorOfTheMonth from './TopPlayCreatorOfTheMonth';
+import TopPlayCreators from './TopPlayCreators';
+import { Watch } from 'react-loader-spinner';
+import _ from 'lodash';
```
Comment (atapas) [inline src/common/playleaderboard/LeaderBoard.jsx:7]: Do we need to import the entire library?

File: src/common/playleaderboard/LeaderBoard.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                <div>
+                  <TopPlayCreators topPlayCreators={top10Contributors} />
+                </div>
+              </>
+            )}
+          </div>
+        </div>
+      ) : (
+        <div
+          style={{
```
Comment (atapas) [inline src/common/playleaderboard/LeaderBoard.jsx:92]: Can we take out the inline style to the CSS file?

File: src/common/playleaderboard/leaderBoard.css [hunk 3]
```diff
@@ -0,0 +1,18 @@
+.table-cell {
```
Comment (atapas) [inline src/common/playleaderboard/leaderBoard.css:1]: The css class names are very generic.. may conflict.

**Review by atapas — CHANGES_REQUESTED** [review]
@NagarjunShroff Initial review comments.

**Comment by atapas** [comment]
@NagarjunShroff The build consists a bunch of the linting errors. please fix them.

**Comment by atapas** [comment]
Hello @NagarjunShroff got a chance to check the comments? Also please resolve conflicts.

File: src/common/playleaderboard/leaderBoard.css (hunk 3 above)
Comment (NagarjunShroff) [reply src/common/playleaderboard/leaderBoard.css]: changed.

File: src/common/playleaderboard/LeaderBoard.jsx (hunk 2 above)
Comment (NagarjunShroff) [reply src/common/playleaderboard/LeaderBoard.jsx]: done

File: src/common/playleaderboard/LeaderBoard.jsx (hunk 1 above)
Comment (NagarjunShroff) [reply src/common/playleaderboard/LeaderBoard.jsx]: no - Fixed

**Comment by atapas** [comment]
@NagarjunShroff this build fails due to this error:

image

**Comment by atapas** [comment]
Hi @NagarjunShroff Here are some tiny responsiveness issues I am observing.

- The images in table gets cut

image

- This is on a higher resolution screen. The left and right side is a far apart. It creates an impression lot of real estate waste. Can we provide fix such a way that they auto adjust with the gap between the columns?

image

**Comment by atapas** [comment]
@NagarjunShroff 

I also wanted to ask:

- Do you have details to link the numbers to the list of plays for all the creators on this page? Or do you suggest that to take as a separate issue?

**Comment by NagarjunShroff** [comment]
> @NagarjunShroff
> 
> I also wanted to ask:
> 
> * Do you have details to link the numbers to the list of plays for all the creators on this page? Or do you suggest that to take as a separate issue?

Lets take that up as separate issue.

**Comment by atapas** [comment]
> > @NagarjunShroff
> > I also wanted to ask:
> > 
> > * Do you have details to link the numbers to the list of plays for all the creators on this page? Or do you suggest that to take as a separate issue?
> 
> Lets take that up as separate issue.

Please create an issue once we merge this one.

**Comment by NagarjunShroff** [comment]
> Hi @NagarjunShroff Here are some tiny responsiveness issues I am observing.
> 
> * The images in table gets cut
> 
> image
> 
> * This is on a higher resolution screen. The left and right side is a far apart. It creates an impression lot of real estate waste. Can we provide fix such a way that they auto adjust with the gap between the columns?
> 
> image

@atapas Fixed - please test it.

**Review by atapas — APPROVED** [review]

**Comment by atapas** [comment]
LGTM

**Pull request merged**